### PR TITLE
ENH: Adding option to skip randomization of seed. + other fixes

### PR DESCRIPTION
**Pull request by EmmaRenauld**
- Adding an option to the _dev tracking to have fixed seeds. This allows to create a figure like this:
     
step_size05

- Changing the --discard_last_point option to make it the default choice (as discussed with Gab and other; this makes it more like Dipy)

- Fixing the way the numpy generator and seed are used, as per the[ numpy doc](https://numpy.org/doc/stable/reference/random/generated/numpy.random.seed.html). If you think it's not the best way to do it, I can just delete the last commit.

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: scripts/scil_compute_local_tracking_dev.py [hunk 1]
```diff
@@ -113,13 +113,20 @@ def _build_arg_parser():
...
-             "of \nthe streamline. Default: append them. This is the default "
+        '--keep_last_out_point', action='store_true',
+        help="If set, keep the last point (once out of the tracking mask) "
+             "of \nthe streamline. Default: discard them. This is the default "
              " in \nDipy too. Note that points obtained after an invalid "
              "direction \n(based on the propagator's definition of invalid; "
              "ex when \nangle is too sharp of sh_threshold not reached) are "
-             "never added.")
+             "never added.\n"
+             "REMARK: Our results (our endpoints) seem to differ from dipy's. "
```
Comment (CHrlS98) [inline scripts/scil_compute_local_tracking_dev.py:123]: I think I would prefer not to write `todos` in the script help, but it's just a matter of personal taste.

**Review by CHrlS98 — COMMENTED** [review]
I tested and it seems to work properly.

However, I think this would be more useful if, instead of disabling jittering the seeding positions, we added a `--nb_repeats` arguments telling how many times we want to track from each seed. The behaviour you want to show could still happen, but it would generalize better and might find other applications than only for teaching purposes?

File: scripts/scil_compute_local_tracking_dev.py [hunk 2]
```diff
@@ -113,13 +113,20 @@ def _build_arg_parser():
...
+             "This should be investigated.")
+    track_g.add_argument(
+        "--do_not_randomize_seed_positions", action="store_true",
```
[the remark this answers is not in the document]
Comment (EmmaRenauld) [reply scripts/scil_compute_local_tracking_dev.py:126]: Changed for no_seed_displacement. Ok?

File: scripts/scil_compute_local_tracking_dev.py [hunk 3]
```diff
@@ -113,13 +113,20 @@ def _build_arg_parser():
...
+        help="By default, seed position is moved randomly inside the voxel. "
```
[the remark this answers is not in the document]
Comment (EmmaRenauld) [reply scripts/scil_compute_local_tracking_dev.py:127]: Changed.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by CHrlS98** [comment]
> > However, I think this would be more useful if, instead of disabling jittering the seeding positions, we added a `--nb_repeats` arguments telling how many times we want to track from each seed. The behaviour you want to show could still happen, but it would generalize better and might find other applications than only for teaching purposes?
> 
> That is nice, but right now it seems complicated for me. We would have to remember how many times we called get_next_pos.

Wouldn't we just need to return the same seed `n_repeat` times ? This just changes the start position of a streamline, no?

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by EmmaRenauld** [comment]
Should be good!

File: scilpy/tracking/tests/notes.txt
Comment (CHrlS98) [inline scilpy/tracking/tests/notes.txt:1]: I would like to have @arnaudbore 's input on this file.

**Review by CHrlS98 — COMMENTED** [review]
I would like to validate that the note.txt file in tests is at the right place, but otherwise, everything looks good to me.

File: scilpy/tracking/tests/notes.txt
Comment (arnaudbore) [reply scilpy/tracking/tests/notes.txt:1]: Why not using a regular test file, add this info inside as well as a dummy test that passes.

Comment (EmmaRenauld) [reply scilpy/tracking/tests/notes.txt:1]: I agree. It's just that I wasn't sure how to proceed. Do a create a test_tracker and test_propagator and copy the same text in both?

Comment (arnaudbore) [reply scilpy/tracking/tests/notes.txt:1]: I agree :+1:

Comment (EmmaRenauld) [reply scilpy/tracking/tests/notes.txt:1]: Done

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Review by CHrlS98 — APPROVED** [review]

**Pull request merged**